### PR TITLE
Contruction wood dev and new release needed for forestry paper

### PR DESCRIPTION
**Pull request by abhimishr**
## :bird: Purpose of this PR :bird:

This PR brings the construction wood code development + work done for forestry paper to the main develop branch. These changes are needed to have an intermediate release of the model for submission of the forestry paper (https://doi.org/10.5194/gmd-2021-76)

## :wrench: Checklist for PR creator :wrench:

- [x] Labeling pull request correctly [from the label list](https://github.com/magpiemodel/magpie/labels).

* Low risk : Simple bugfixes (missing files, updated documentation, typos) or Start/output scripts
* Medium risk : New realization / Changes to existing realization / Other changes which don't modify default.cfg
* High risk : New input files (if cfg$input is changed in default.cfg) / Modification to core model (eg. changes in equations, calculations, introduction of new sets etc.) / Other changes in default.cfg

- [x] Providing additional information based on PR label

* Low risk : No new model run needed.
* Medium risk : Default run based on the current version of the fork from which PR is made
* High risk
  * Default run from the current develop branch
  * Default run based on the current version of the fork from which PR is made

- [x] :chart_with_downwards_trend: Performance loss/gain from current default behavior :chart_with_upwards_trend:
  * Current develop branch default :37m51s
  * This PR's default : 30m52s

- [x] Added changes to `CHANGELOG.md`
- [x] Compilation check (model starts without compilation errors - use `gams main.gms action=c` in model folder for testing).
- [x] No hard coded numbers and cluster/country/region names.
- [x] The new code doesn't contain declared but unused parameters or variables.
- [x] Where relevant, In-code comments added including documentation comments.
- [x] Made sure that documentation created with [`goxygen`](https://github.com/pik-piam/goxygen) is okay (use `goxygen::goxygen()` for testing).
- [x] Changes to [`magpie4`](https://github.com/pik-piam/magpie4) R library for post processing of model output (ideally backward compatible).
- [x] Self-review of my own code.
- [x] For high risk runs: validation of major model indicators - Land-use, emissions, food prices, Tau.  %Delete this line in case it is not a high risk run%

## :rotating_light: Checklist for RSE reviewer :rotating_light:

- [x] PR is labeled correctly.
- [x] `CHANGELOG` is updated correctly
- [x] No hard coded numbers and cluster/country/region names.
- [x] No unnecessary increase in module interfaces
- [x] All required updates in interfaces (if any) have been properly adressed in the module contracts
- [x] In-code comments and documentation comments are satisfactory.
- [x] model behavior/performance is satisfactory.
- [x] Requested changes (if any) were applied correctly

## :rotating_light: Checklist for MAgPIE reviewer :rotating_light:

- [x] PR is labeled correctly.
- [x] `CHANGELOG` is updated correctly
- [x] No hard coded numbers and cluster/country/region names.
- [x] Changes to the model are scientifically sound
- [x] In-code comments and documentation comments are satisfactory.
- [x] model behavior/performance is satisfactory.
- [x] Requested changes (if any) were applied correctly

### CODE REVIEW

File: scripts/start/projects/forestry.R [hunk 1]
```diff
@@ -23,49 +23,100 @@ source("scripts/start_functions.R")
...
+                  }
+                }
+              }
+          }
+        }
+     }
+  }
 }
 
+#for(i in names(all_configs)){
```
Comment (tscheypidi) [inline scripts/start/projects/forestry.R:114]: please avoid committing out-commented code. Include it with a switch instead if it should be optional

File: scripts/start/projects/forestry_lpjml_addon.R [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                  }
+                }
+              }
+          }
+        }
+     }
+  }
+}
+
+#for(i in names(all_configs)){
```
Comment (tscheypidi) [inline scripts/start/projects/forestry_lpjml_addon.R:116]: please avoid committing out-commented code. Include it with a switch instead if it should be optional

**Review by tscheypidi — APPROVED** [review]
just the minor comment to please not commit out-commented code. Besides that it looks ok from RSE side

File: scripts/start/projects/forestry_lpjml_addon.R (hunk 2 above)
Comment (abhimishr) [reply scripts/start/projects/forestry_lpjml_addon.R]: Removed unused code

File: config/default.cfg [hunk 3]
```diff
@@ -616,6 +617,13 @@ cfg$gms$c32_incr_rate <- "h5s2l1"             # def = "h5s2l1"
 # *      All plantations are harvested at rotation age. Plantation establishment is endogenous.
 cfg$gms$s32_hvarea = 0 # def = 0
 
+# Type of rotation length selection criteria
+# * ("max_marginal_increment")      = Harvesting when the average annual increment is maximum
```
Comment (flohump) [inline config/default.cfg:621]: This is confusing. Change to "mean_annual_increment" ?

File: modules/32_forestry/dynamic_feb21/preloop.gms [hunk 4]
```diff
@@ -30,9 +30,24 @@ p32_IGR(t_all,j,ac) =   (p32_carbon_density_ac_marg(t_all,j,ac)/p32_carbon_densi
...
+  p32_rot_flg(t_all,j,ac) = 1$(p32_carbon_density_ac_marg(t_all,j,ac) - p32_carbon_density_ac_marg(t_all,j,ac-1) >  0)
+                          + 0$(p32_carbon_density_ac_marg(t_all,j,ac) - p32_carbon_density_ac_marg(t_all,j,ac-1) <= 0);
+  display "Rotation lengths are calculated based on maximizing CAI in this run.";
+$endif
+
+$ifthen "%c32_rot_calc_type%" == "max_marginal_increment"
+  p32_avg_increment(t_all,j,ac) = pm_carbon_density_ac_forestry(t_all,j,ac,"vegc") / ((ord(ac)+1)*5);
+  p32_rot_flg(t_all,j,ac) = 1$(p32_carbon_density_ac_marg(t_all,j,ac) - p32_avg_increment(t_all,j,ac) >  0)
+                          + 0$(p32_carbon_density_ac_marg(t_all,j,ac) - p32_avg_increment(t_all,j,ac) <= 0);
+  display "Rotation lengths are calculated based on maximizing increment in this run.";
```
Comment (flohump) [inline modules/32_forestry/dynamic_feb21/preloop.gms:50]: maximizing MAI

File: modules/32_forestry/dynamic_feb21/preloop.gms [hunk 5]
```diff
@@ -30,9 +30,24 @@ p32_IGR(t_all,j,ac) =   (p32_carbon_density_ac_marg(t_all,j,ac)/p32_carbon_densi
 $ifthen "%c32_interest_rate%" == "regional"
   p32_rot_flg(t_all,j,ac) = 1$(p32_IGR(t_all,j,ac) - sum(cell(i,j),pm_interest("y1995",i)) >  0)
                           + 0$(p32_IGR(t_all,j,ac) - sum(cell(i,j),pm_interest("y1995",i)) <= 0);
+  display "Rotation lengths are calculated based on maximizing NPV.";
```
Comment (flohump) [inline modules/32_forestry/dynamic_feb21/preloop.gms:33]: Is NPV correct? Or instantaneous growth rate equated to interest rate?

File: modules/73_timber/default/preloop.gms [hunk 6]
```diff
@@ -47,6 +47,36 @@ $endif
 ** p73_timber_demand_gdp_pop is in mio m^3
 ** pm_demand_ext in mio ton DM
 ** Hold constraint beyond 2150 - First every time step gets 2150 values
+**** Extend for Churkina et al 2020 demand scenarios
 pm_demand_ext(t_ext,i,kforestry) = round(p73_timber_demand_gdp_pop("y2150",i,kforestry) * f73_volumetric_conversion(kforestry),3);
 ** overwrite timesteps below 2150 with actual values
 pm_demand_ext(t_all,i,kforestry) = round(p73_timber_demand_gdp_pop(t_all,i,kforestry) * f73_volumetric_conversion(kforestry),3);
+
+p73_demand_modifier(t_all) = 0;
+p73_fraction(t_all) = s73_expansion/80;
```
Comment (flohump) [inline modules/73_timber/default/preloop.gms:56]: 80 = 2100 - 2020
You can use m_year(t) for this, this works in preloop
m_year("y2100") - m_year("y2020")
Maybe calculate it once and save it in scalar?

File: scripts/start/projects/forestry_lpjml_addon.R [hunk 7]
```diff
@@ -0,0 +1,114 @@
+# |  (C) 2008-2021 Potsdam Institute for Climate Impact Research (PIK)
```
Comment (flohump) [inline scripts/start/projects/forestry_lpjml_addon.R:1]: Can be deleted?

Comment (flohump) [reply scripts/start/projects/forestry_lpjml_addon.R:1]: Is this start script needed for your release?

**Review by flohump — CHANGES_REQUESTED** [review]
Overall fine. Some minor changes are needed, which do not affect the model outcome.

File: config/default.cfg (hunk 3 above)
Comment (abhimishr) [reply config/default.cfg]: Changed via 2fdde79

File: modules/32_forestry/dynamic_feb21/preloop.gms (hunk 4 above)
Comment (abhimishr) [reply modules/32_forestry/dynamic_feb21/preloop.gms]: Changed via 2fdde79

File: modules/32_forestry/dynamic_feb21/preloop.gms (hunk 5 above)
Comment (abhimishr) [reply modules/32_forestry/dynamic_feb21/preloop.gms]: Edited for clarity. Changed via 2fdde79

File: modules/73_timber/default/preloop.gms (hunk 6 above)
Comment (abhimishr) [reply modules/73_timber/default/preloop.gms]: Redid the calculation for clean steps. Changed via 2fdde79

File: scripts/start/projects/forestry_lpjml_addon.R (hunk 7 above)
Comment (abhimishr) [reply scripts/start/projects/forestry_lpjml_addon.R:1]: I'll keep it (but not needed for release). It stays in the project folder

**Review by flohump — APPROVED** [review]

**Pull request merged**